src/qibolab/designs/qmrs.py

```diff
@@ -0,0 +1,160 @@
+        Channel("L3-12").local_oscillator = self.local_oscillators[3]
+        Channel("L3-13").local_oscillator = self.local_oscillators[4]
+        Channel("L3-14").local_oscillator = self.local_oscillators[4]
+
+        # Set default LO parameters in the channel
+        Channel("L3-25_a").lo_frequency = 7_300_000_000
+        Channel("L3-25_b").lo_frequency = 7_900_000_000
+        Channel("L3-15").lo_frequency = 4_700_000_000
+        Channel("L3-11").lo_frequency = 4_700_000_000
+        Channel("L3-12").lo_frequency = 5_600_000_000
+        Channel("L3-13").lo_frequency = 6_500_000_000
+        Channel("L3-14").lo_frequency = 6_500_000_000
+
+        Channel("L3-25_a").lo_power = 18.0
+        Channel("L3-25_b").lo_power = 15.0
+        Channel("L3-15").lo_power = 16.0
+        Channel("L3-11").lo_power = 16.0
+        Channel("L3-12").lo_power = 16.0
+        Channel("L3-13").lo_power = 16.0
+        Channel("L3-14").lo_power = 16.0
+
+        # drop the local oscillators to avoid connecting when simulation is used
+        if simulation_duration is not None:
+            self.local_oscillators = []
+
+    def connect(self):
+        self.opx.connect()
+        if not self.is_connected:
+            for lo in self.local_oscillators:
+                try:
+                    log.info(f"Connecting to instrument {lo}.")
+                    lo.connect()
+                except Exception as exception:
+                    raise_error(
+                        RuntimeError,
+                        f"Cannot establish connection to {lo} instruments. Error captured: '{exception}'",
+                    )
+            self.is_connected = True
+
+    def setup(self, qubits):
+        # setup QM (HARDCODED values for configuration)
+        self.opx.setup(qubits, time_of_flight=280, smearing=0)
+
+        # set LO frequencies
+        for qubit in qubits:
+            for channel in [qubit.readout, qubit.drive]:
+                if channel is not None and channel.local_oscillator is not None:
+                    # set LO frequency
+                    lo = channel.local_oscillator
+                    frequency = channel.lo_frequency
+                    if lo.is_connected:
+                        lo.setup(frequency=frequency, power=channel.lo_power)
+                    else:
+                        log.warn(f"There is no connection to {lo}. Frequencies were not set.")
+
+    def start(self):
+        self.opx.start()
+        if self.is_connected:
+            for lo in self.local_oscillators:
+                lo.start()
+
+    def stop(self):
+        if self.is_connected:
+            for lo in self.local_oscillators:
+                lo.stop()
+        self.opx.stop()
+
+    def disconnect(self):
+        if self.is_connected:
+            self.opx.disconnect()
+            for lo in self.local_oscillators:
+                lo.disconnect()
+            self.is_connected = False
```

Comment:
Yes and no. The design of having controller electronics (in this case the OPXs) mixed with local oscillators is quite common but there are cases that are different. I would not move this to the abstract class mainly because we want to accommodate instrument designs that do not use LOs.

But we should keep in mind that this design could be used in other cases. This will be more clear if we refactor different instruments to this format. For sure I would rename `opx` to something else because this term is very specific to the QM instruments.

Reply:
Done as we now have two generic designs (with and without mixers).